lib/Common/Common/NumberUtilities_strtod.cpp

```diff
@@ -2525,10 +2525,11 @@ BOOL Js::NumberUtilities::FNonZeroFiniteDblToStr(double dbl, _In_range_(2, 36) i
     // powers of 2, but not for other radixes.
     // REVIEW : round?
     wExp2 = (int)((Js::NumberUtilities::LuHiDbl(dbl) & 0x7FF00000) >> 20) - 0x03FF;
+    wExp10 = (int)floor(log10(dbl));
     maxOutDigits = g_rgcchSig[radix];
     __analysis_assume(maxOutDigits > 0);
 
-    if (wExp2 < -60 || wExp2 > 60)
+    if (wExp10 < -21 || wExp10 > 21)
```

Comment:
Exp2 is already calculated above. Could we use that for comparison instead of calculating expensive Exp10?